From eda0eec3e942a7e1467de105976d88e3e399c8c1 Mon Sep 17 00:00:00 2001
From: HuaWen CHEN <quic_huawenc@quicinc.com>
Date: Thu, 23 Nov 2023 19:47:48 +0800
Subject: [PATCH] [PATCH] mwan3: change for default route

to support only one default route in main table:
1. netifd will add default route of non-default backhaul to customized route table
2. mwan3 should not handle default route deleted event to avoid race-condition.
   scenario:
   during backhaul switch,
   netifd delete the default route in main table and then add same in customzied table.
   mwan3 may delete route in customized table when receive the delete event on main table.
---
 net/mwan3/files/usr/sbin/mwan3rtmon | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/net/mwan3/files/usr/sbin/mwan3rtmon b/net/mwan3/files/usr/sbin/mwan3rtmon
index d8ccffeb04..b018cc7c48 100755
--- a/net/mwan3/files/usr/sbin/mwan3rtmon
+++ b/net/mwan3/files/usr/sbin/mwan3rtmon
@@ -112,6 +112,11 @@ mwan3_rtmon_route_handle()
 			return
 		fi
 
+		if [ $action = "del" ] && [[ "$route_line" == default* ]]; then
+			LOG debug "skipping deleted default route for table $tid - '$route_line'"
+			return
+		fi
+
 		network_get_device device "$iface"
 		LOG debug "adjusting route $device: '$IP route $action table $tid $route_line'"
 		error=$($IP route "$action" table $tid $route_line 2>&1)||
-- 
2.40.0.windows.1

